### PR TITLE
added solution for spring security part 2

### PR DESCRIPTION
**Pull request by Basilius-Third**

### CODE REVIEW

File: src/main/java/mate/academy/spring/config/SecurityConfig.java [hunk 1]
```diff
@@ -1,28 +1,52 @@
...
-                .passwordEncoder(getEncoder())
-                .withUser("1@123").password(getEncoder().encode("123")).roles("USER");
+    public void configure(AuthenticationManagerBuilder auth) throws Exception {
+        auth.userDetailsService(userDetailsService).passwordEncoder(passwordEncoder);
     }
 
     protected void configure(HttpSecurity http) throws Exception {
         http
                 .authorizeRequests()
-                .anyRequest().authenticated()
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/spring/config/SecurityConfig.java:25]: does it work without this line after `.antMatchers`?

File: src/main/java/mate/academy/spring/service/CustomUserDetailsService.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    @Autowired
+    private UserService userService;
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/spring/service/CustomUserDetailsService.java:16]: please use constructor autowiring

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/spring/config/SecurityConfig.java (hunk 1 above)
Comment (Basilius-Third) [reply src/main/java/mate/academy/spring/config/SecurityConfig.java:25]: It does, but I'll return this line if you'll say so

File: src/main/java/mate/academy/spring/config/SecurityConfig.java [hunk 3]
```diff
@@ -1,39 +1,59 @@
...
+                .antMatchers(HttpMethod.PUT, "/movie-sessions/{id}").hasRole(ADMIN)
+                .antMatchers(HttpMethod.DELETE, "/movie-sessions/{id}").hasRole(ADMIN)
+                .antMatchers(HttpMethod.GET, "/orders").hasRole(USER)
+                .antMatchers(HttpMethod.POST, "/orders/complete").hasRole(USER)
+                .antMatchers(HttpMethod.POST, "/shopping-carts/movie-sessions").hasRole(USER)
+                .antMatchers(HttpMethod.GET, "/shopping-carts/by-user").hasRole(USER)
+                .antMatchers(HttpMethod.GET, "/users/by-email").hasRole(ADMIN)
+                .and()
                 .formLogin()
                 .permitAll()
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/spring/config/SecurityConfig.java:53]:
```suggestion
                .authorizeRequests()
                .antMatchers(HttpMethod.POST, "/register").permitAll()
                .antMatchers(HttpMethod.GET, "/cinema-halls").hasAnyRole(USER, ADMIN)
                .antMatchers(HttpMethod.POST, "/cinema-halls").hasRole(ADMIN)
                .antMatchers(HttpMethod.GET, "/movies").hasAnyRole(USER, ADMIN)
                .antMatchers(HttpMethod.POST, "/movies").hasRole(ADMIN)
                .antMatchers(HttpMethod.GET, "/movie-sessions/available").hasAnyRole(USER, ADMIN)
                .antMatchers(HttpMethod.GET, "/movie-sessions/{id}").hasAnyRole(USER, ADMIN)
                .antMatchers(HttpMethod.POST, "/movie-sessions").hasRole(ADMIN)
                .antMatchers(HttpMethod.PUT, "/movie-sessions/{id}").hasRole(ADMIN)
                .antMatchers(HttpMethod.DELETE, "/movie-sessions/{id}").hasRole(ADMIN)
                .antMatchers(HttpMethod.GET, "/orders").hasRole(USER)
                .antMatchers(HttpMethod.POST, "/orders/complete").hasRole(USER)
                .antMatchers(HttpMethod.POST, "/shopping-carts/movie-sessions").hasRole(USER)
                .antMatchers(HttpMethod.GET, "/shopping-carts/by-user").hasRole(USER)
                .antMatchers(HttpMethod.GET, "/users/by-email").hasRole(ADMIN)
                .anyRequest().authenticated()
                .and()
                .formLogin()
                .permitAll()
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**